### PR TITLE
feat(ui): add mouse selection, copy-to-clipboard, and click-to-open links

### PR DESCRIPTION
**Pull request by WilliamAGH**
## Summary

Adds interactive mouse features for the chat history including text selection with copy-to-clipboard, click-to-open links, and visual feedback. Also includes composer improvements for better message typing and deterministic build setup.

## Features

- Mouse text selection: Drag to select text in chat history with visual highlighting
- Copy to clipboard: Release mouse selection to copy selected text (shows COPIED status)
- Click-to-open links: Click URLs in chat history to open in browser (shows OPENED status)
- Mouse cursor shapes: Pointer cursor over links, text cursor over selectable content
- Composer styling: Enhanced prompt styling with theme colors, better placeholder appearance
- Paste handling: Support for pasting content into composer with proper message routing
- Keyboard shortcuts: Ctrl+J for newline, / for command palette, Esc to quit
- Makefile test target: Convenient test command for running tests

## Bug Fixes

- Fix composer view to preserve explicit newlines created via Shift+Enter (previously dropped trailing padding incorrectly)

## Technical Improvements

- Extract URL utilities to UrlUtil.java for link detection and normalization
- Extract composer rendering to ComposerViewLines.java for cleaner separation
- Refactor MouseSelectionController into focused methods (press/drag/release handling)
- Enable mouse target cursor support in Program initialization

## Build System & CI/CD

- Add Gradle Toolchains configuration with Temurin JDK 25 for reproducible builds
- Add .tool-versions for mise/asdf version manager support
- Add GitHub Actions build workflow for automated testing
- Update release workflow to Actions v6 and stable Java 25
- Add TUI4J_SOURCE environment variable for explicit local/maven dependency selection
- Upgrade JUnit to 6.0.2, add Mockito 5.21.0, update tui4j snapshot
- Quiet Gradle test output for cleaner CI logs

## Documentation

- Add comprehensive development guide (docs/development.md)
- Add contribution guidelines (CONTRIBUTING.md)
- Add code change policy contract (docs/contracts/code-change.md)
- Update README with development setup instructions

## Code Organization

- Enforce 350 LOC limit per file (LOC1 rule) via AGENTS.md
- Establish monolith prevention guidelines (MO1 rule)

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -5,13 +5,27 @@ plugins {
...
-val tui4jLocalRequested = listOf(
-    findProperty("tui4jLocal")?.toString(),
-    System.getenv("TUI4J_LOCAL"),
-).any { it == "true" }
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(25)
+    }
+}
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply build.gradle.kts]: **Analysis: Acknowledged**

You're correct that the hardcoded path in `tui4jLocalJarCandidates` is user-specific. However, this is intentional design:

1. **TUI4J_LOCAL_PATH environment variable** already provides the recommended override mechanism for contributors to set their own paths
2. The hardcoded paths serve as convenient defaults for the primary development environment
3. The list is prioritized with the env var first, so it takes precedence when set

The existing setup allows:
```bash
TUI4J_LOCAL_PATH=/path/to/your/tui4j.jar make run
```

This is documented in the code and provides the flexibility needed for contributors without requiring changes to the build file.

File: docs/development.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+#### IntelliJ doesn't pick up Java version
+- Open IntelliJ settings → Build, Execution, Deployment → Gradle
+- Set "Gradle JVM" to "Use JAVA_HOME"
+- Set "Project SDK" to Temurin 25 (or refresh if auto-detected)
+
+#### CI build fails but local works
+- Check CI logs for Java version (e.g., `java -version`)
+- Ensure your local patch matches
+- If CI is on 25.0.4 and you're on 25.0.3, bump locally and re-test
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply docs/development.md]: **Suggestion Implemented ✅**

Fixed all heading level violations in the Troubleshooting section. Changed H4 headings (####) to H3 (###) to satisfy MD001 - headings must only increment by one level at a time under the H2 "Troubleshooting" section.

**Changes:**
- `#### "gradle: command not found"` → `### "gradle: command not found"`
- `#### Gradle downloads JDK every time` → `### Gradle downloads JDK every time`
- `#### IntelliJ doesn't pick up Java version` → `### IntelliJ doesn't pick up Java version`
- `#### CI build fails but local works` → `### CI build fails but local works`

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+- Uses same Temurin Java 25 + latest Actions versions
+- Builds distribution ZIP and updates Homebrew tap
+
+### Gradle Configuration
+
+#### `settings.gradle.kts` (Foojay Resolver)
+
+```kotlin
+plugins {
+    id("org.gradle.toolchains.foojay-resolver-convention") version "0.9.0"
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply docs/development.md]: Suggestion implemented. Updated version to 1.0.0 to match settings.gradle.kts.

File: docs/development.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+#### `build.gradle.kts` (Toolchain Vendor)
+
+```kotlin
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(25)
+        vendor = JvmVendorSpec.ADOPTOPENJDK
+    }
+}
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply docs/development.md]: Suggestion implemented. Removed vendor spec to align with build.gradle.kts.

File: src/main/java/com/williamcallahan/chatclient/ui/UrlUtil.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        if (URL_PATTERN.matcher(t).matches()) {
+            return t.contains("://") ? t : "https://" + t;
+        }
+        return null;
+    }
+
+    private static String stripPunctuation(String t) {
+        return t.replaceAll("^[\"'\\[\\(<{`]+|[\"'\\]\\)>}`.,]+$", "");
+    }
+}
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply src/main/java/com/williamcallahan/chatclient/ui/UrlUtil.java]: Added UrlUtilTest.java with comprehensive test cases for isPotentialUrl and normalizeUrl, covering edge cases like punctuation wrapping, IP addresses, and various URL formats.

File: src/main/java/com/williamcallahan/chatclient/ui/ComposerViewLines.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    private static int findAnsiSequenceEnd(String text, int pos) {
+        int end = pos + 2;
+        while (end < text.length() && text.charAt(end) != 'm') {
+            end++;
+        }
+        return (end < text.length()) ? end + 1 : pos;
+    }
+}
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply src/main/java/com/williamcallahan/chatclient/ui/ComposerViewLines.java]: Added ComposerViewLinesTest.java covering single-line, multi-line (with prompt alignment), empty input, and ANSI sequence handling scenarios.

File: docs/contracts/code-change.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
+---
+title: "Code change policy contract"
+usage: "Use whenever creating/modifying files: where to put code, when to create new types, and how to stay SRP compliant"
+description: "Evergreen contract for change decisions (new file vs edit), repository structure, and library architecture; references rule IDs in `AGENTS.md`"
+---
+
+# Code Change Policy Contract
+
+See `AGENTS.md` ([LOC1a], [MO1a], [DEP1]).
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply docs/contracts/code-change.md]: Suggestion implemented. Corrected rule IDs to [LOC1], [MO1], [DEP1].
